### ISSUE TITLE
DepthFormer results on NYU dataset 

### ISSUE THREAD

**Issue by mshaheryarmalik** [zhyever/Monocular-Depth-Estimation-Toolbox#34]
Hello,

I trained DepthFormer model on NYU dataset after processing the raw data from [other work](https://github.com/cleinc/bts/blob/master/utils/extract_official_train_test_set_from_mat.py) and made few modifications to code such as:

- Added GradientCumulativeOptimizerHook in optimizer_config ([DepthFormer config](https://github.com/zhyever/Monocular-Depth-Estimation-Toolbox/blob/792a35ca413cc7e435c256717b8bfc2140a7098d/configs/depthformer/depthformer_swint_w7_nyu.py#L60)) to handle larger batch size I tired with 16)
- Batch size `samples_per_gpu=4, workers_per_gpu=4` [here](https://github.com/zhyever/Monocular-Depth-Estimation-Toolbox/blob/792a35ca413cc7e435c256717b8bfc2140a7098d/configs/depthformer/depthformer_swint_w7_nyu.py#L33)
- Added this to checkpoint_config `create_symlink=False` [here](https://github.com/zhyever/Monocular-Depth-Estimation-Toolbox/blob/792a35ca413cc7e435c256717b8bfc2140a7098d/configs/depthformer/depthformer_swint_w7_nyu.py#L63)

However, when I try to predict depth with newly trained model, it just gives strange results that if you can explain. I have attached _training logs_ for your reference and one prediction example from newly trained model below.

![data_nyu_bathroom_rgb_00045](https://user-images.githubusercontent.com/25252447/178583402-dbfad8c2-aa93-45e4-99de-c43e5e791132.jpg)

P.S: I was able to get good results with BTS with same code modifications.
[20220708_175744.log](https://github.com/zhyever/Monocular-Depth-Estimation-Toolbox/files/9096334/20220708_175744.log)

**Comment by zhyever** [comment]
This is related to the unsolved issue of convergence. Some discussions can be seen in Issue #20, #32. As you can see from the log files, the sigloss is not decreasing and retain the value of about 3. 

While I have tried several methods to solve this issue, they are futile and the problem exists. Now, the simplest way is to **_run the training script several times_**. Whenever you see that the loss is less than 3 (can decrease to 0.x) in the first several printed log info, it means that the model can be successfully trained in the following training stage.

Refer to Issue #20, #32 for more details.

**Comment by zhyever** [comment]
Close for now.
